overlord/devicestate/firstboot.go

```diff
@@ -41,6 +41,35 @@ import (
 
 var errNothingToDo = errors.New("nothing to do")
 
+func installSeedSnap(st *state.State, sn *snap.SeedSnap, flags snapstate.Flags) (*state.TaskSet, error) {
+	if sn.Classic {
+		flags.Classic = true
+	}
+	if sn.DevMode {
+		flags.DevMode = true
+	}
+
+	path := filepath.Join(dirs.SnapSeedDir, "snaps", sn.File)
+
+	var sideInfo snap.SideInfo
+	if sn.Unasserted {
+		sideInfo.RealName = sn.Name
+	} else {
+		si, err := snapasserts.DeriveSideInfo(path, assertstate.DB(st))
```

Comment:
shouldn't you try to derive it anyway, and only error out if it fails and isn't unasserted?